code/core/src/manager/components/sidebar/ChecklistModule.tsx

```diff
@@ -0,0 +1,322 @@
+import { useStorybookApi } from 'storybook/manager-api';
+import { keyframes, styled } from 'storybook/theming';
+
+import { Optional } from '../Optional/Optional';
+import { Particles } from '../Particles/Particles';
+import { TextFlip } from '../TextFlip';
+import type { ChecklistItem } from './useChecklist';
+import { useChecklist } from './useChecklist';
+
+const fadeScaleIn = keyframes`
+  from {
+    opacity: 0;
+    transform: scale(0.7);
+  }
+  to {
+    opacity: 1;
+    transform: scale(1);
+  }
+`;
+
+const expand = keyframes`
+  from {
+    transform: scaleX(0);
+  }
+  to {
+    transform: scaleX(1);
+  }
+`;
+
+const useTransitionArray = <K,>(array: K[], subset: K[], options: TransitionMapOptions<K>) => {
+  const { setItem, toggle, stateMap } = useTransitionMap<K>({
+    allowMultiple: true,
+    mountOnEnter: true,
+    unmountOnExit: true,
+    preEnter: true,
+    ...options,
+  });
+
+  useEffect(() => {
+    array.forEach((task) => setItem(task));
+  }, [array, setItem]);
+
+  useEffect(() => {
+    array.forEach((task) => toggle(task, subset.includes(task)));
+  }, [array, subset, toggle]);
+
+  return Array.from(stateMap);
+};
+
+const CollapsibleWithMargin = styled(Collapsible)(({ collapsed }) => ({
+  marginTop: collapsed ? 0 : 16,
+}));
+
+const HoverCard = styled(Card)({
+  '&:hover #checklist-module-collapse-toggle': {
+    opacity: 1,
+  },
+});
+
+const CollapseToggle = styled(ListboxButton)({
+  opacity: 0,
+  transition: 'opacity var(--transition-duration, 0.2s)',
+  '&:focus, &:hover': {
+    opacity: 1,
+  },
+});
+
+const ProgressCircle = styled(ProgressSpinner)(({ theme }) => ({
+  color: theme.color.secondary,
+}));
+
+const Checked = styled(StatusPassIcon)(({ theme }) => ({
+  padding: 1,
+  borderRadius: '50%',
+  background: theme.color.positive,
+  color: theme.background.content,
+  animation: `${fadeScaleIn} 500ms forwards`,
+}));
+
+const ItemLabel = styled.span<{ isCompleted: boolean; isSkipped: boolean }>(
+  ({ theme, isCompleted, isSkipped }) => ({
+    position: 'relative',
+    margin: '0 -2px',
+    padding: '0 2px',
+    color: isSkipped
+      ? theme.color.mediumdark
+      : isCompleted
+        ? theme.base === 'dark'
+          ? theme.color.positive
+          : theme.color.positiveText
+        : theme.color.defaultText,
+    transition: 'color 500ms',
+  }),
+  ({ theme, isSkipped }) =>
+    isSkipped && {
+      '&:after': {
+        content: '""',
+        position: 'absolute',
+        top: '50%',
+        left: 0,
+        width: '100%',
+        height: 1,
+        background: theme.color.mediumdark,
+        animation: `${expand} 500ms forwards`,
+        transformOrigin: 'left',
+      },
+    }
+);
+
+const title = (progress: number) => {
+  switch (true) {
+    case progress < 25:
+      return 'Getting started';
+    case progress < 50:
+      return 'Making progress';
+    case progress < 75:
+      return 'Getting close';
+    default:
+      return 'Almost there';
+  }
+};
```

Comment:
Might as well make this a react component, IMHO

Reply:
Unless we want to throw icons or emoji in there, I see no reason to use a less efficient solution.